File: dataset/avatar/Python/Code/codeforces_242_A.py
x , y , a , b = map ( int , input ( ).split ( ) )
games = [ ]
for i in range ( a , x + 1 ) :
    if ( i <= b ) : continue
    else :
        for j in range ( b , y + 1 ) :
            if i > j :
                games.append ( ( i , j ) )
print ( len ( games ) )
for i in games :
    print ( f"{i[0]} {i[1]}" )

File: output/llama_4_maverick/avatar/algo_based_translation/python/java/successful_translation/codeforces_242_A.java

import java.util.ArrayList;
import java.util.Scanner;

public class codeforces_242_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int x = Integer.parseInt(input[0]);
        int y = Integer.parseInt(input[1]);
        int a = Integer.parseInt(input[2]);
        int b = Integer.parseInt(input[3]);
        ArrayList<String> games = new ArrayList<>();
        for (int i = a; i <= x; i++) {
            if (i <= b) continue;
            else {
                for (int j = b; j <= y; j++) {
                    if (i > j) {
                        games.add(i + " " + j);
                    }
                }
            }
        }
        System.out.println(games.size());
        for (String game : games) {
            System.out.println(game);
        }
    }
}

